src/Containers/CollapsibleAccordion/CollapsibleAccordionItem.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useState, useRef, useLayoutEffect } from 'react';
+import styled from 'styled-components';
+import { AngleRight } from '@styled-icons/fa-solid/AngleRight';
+import { ShoppingCart } from '@styled-icons/remix-line/ShoppingCart';
+import { Mixins } from '../../Utils';
+import { transition } from '../../Utils/Mixins';
+import { Heading } from '../../Text';
+
+export interface CollapsibleAccordionItemProps
+    extends React.HTMLAttributes<HTMLDivElement> {
+    /* Heading text */
+    header: string;
+    /* Month and Year of purchases */
+    purchaseDate: string;
+    /* Total amount of purchases */
+    purchaseCount: number;
+}
+
+export interface SectionDivProps {
+    isActive: boolean;
+    height: number;
+}
+
+export interface HeaderDivProps {
+    isActive: boolean;
+}
+
+export interface AngleIconProps {
+    isActive: boolean;
+}
+
+/**
+     * Logic for the height given to the expanded accordion based on the amount of inside children
+     */
+export const CollapsibleAccordionItem: React.FC<CollapsibleAccordionItemProps> = ({
+    header,
+    purchaseDate,
+    purchaseCount,
+    ...props
+}): React.ReactElement => {
+    const [isActive, setIsActive] = useState(false);
+    const [height, setHeight] = useState(0);
+    const toggleIsActive = (): void => {
+        setIsActive(!isActive);
+    };
+    const bodyRef = useRef<HTMLDivElement>(null);
+    const headerRef = useRef<HTMLDivElement>(null);
+    const { children } = props;
+
+    useLayoutEffect((): void => {
+        const bodyNode = bodyRef.current;
+        const headerNode = headerRef.current;
+        if (bodyNode && headerNode) {
+            if (isActive) {
+                setHeight(bodyNode.clientHeight + headerNode.clientHeight);
+            } else {
+                setHeight(headerNode.clientHeight);
+            }
```

Comment:
Have a variable maintaining the height, then just have ONE setHeight at the end of the function . That will make the code more readable

Reply:
Fixed